### PR TITLE
Minor UI and Alignment fixes 

### LINKED ISSUE #256: Add functionality to delete items from the wishlist section
## Is your feature request related to a problem? Please describe.

Add a user functionality to delete items/courses from the "Wishlist" section on click of that delete button.

## Additional context

wishlist-section

### LINKED ISSUE #303: UI overflow on smaller devices
## Describe the bug

when we open the "peer page" on smaller devices (width between 320 to 360) then the "post something" pop out seems not aligned properly in the center it is slightly right or left based on the screen size, same with the referral page "social share pop-up".

## To Reproduce

1. go the peer page.
2. open the peer page on the smaller device using chrome inspect or use react deployed link. 
3. click on the "post something" option.
4. same for the referral page.

## Expected behavior

aligned properly on center and on web page try to bind in the within the UI space to avoid alignment issues

## Environment (please complete the following information):

- OS: windows
- Browser: chrome

## Additional context

***screenshot 1***
alignment issue

***screenshot 2***
alignemt post something

### PR DESCRIPTION
**Pull request by Abhijay007**
fixes #303

**Describe the changes you've made**

Minor UI and Alignment fixes for smaller width devices for smaller devices, UI gets overflowed in some cases.

**Describe if there is any unusual behavior (Any Warning) of your code(Write `NA` if there isn't)**

`NA`

**Checklist**

- [x] My code follows the code style of this project.
- [x] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

**Provide a Deployed link of route/page that needs to review**

Preview: https://deploy-preview-296--edu-client.netlify.app/peer
 
Referral Page :https://deploy-preview-296--edu-client.netlify.app/referralAndCoupons

### CODE REVIEW

**Comment by sakshi-choudhary** [comment]
@Abhijay007 I think there's an issue. It doesn't delete that particular item I select, it deletes the item at the last.

**Comment by Abhijay007** [comment]
> @Abhijay007 I think there's an issue. It doesn't delete that particular item I select, it deletes the item at the last.

@sakshi-choudhary try to delete the last element, it seems that it is deleting the last item but, it is deleting a particular item. try it as I showed in the above gif. I did a self-analysis too and I think that it works fine. 

if this bug exits, I will try my best to solve it.

**Comment by Abhishek-kumar09** [comment]
I don't know the deploy preview is not working right now in my browser, shows a blank page when I move to this route deploy-preview-296--festive-allen-a7f545.netlify.app/wishlist

Is it same in your case?

**Comment by Abhijay007** [comment]
Yes deploy preview is not working.

**Comment by Abhishek-kumar09** [comment]
Github, rebase

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Why wishlistcard/index.js is deleted?

Please copy paste package-lock.json from the latest repo in your next commit.

File: src/pages/wishListView/index.js [hunk 1]
```diff
@@ -52,13 +117,68 @@ const WishListView = (props) => {
...
+                        title="Checkout Now"
+                        onClick={() => history.push(`/course/${id}`)}
+                      />
+                    </CardContent>
+                  </Card>
+                </Container>
+              </ListItem>
+            );
+          })
+        )}
```
Comment (Abhishek-kumar09) [inline src/pages/wishListView/index.js:181]: WishlistCard is already created for that, it's redundant code. Please try to replace it with WishlistCard and pass props if needed for deletion feature

**Comment by Abhijay007** [comment]
@Abhishek-kumar09 I modified this PR for UI and alignment changes as this issue (#256) is already closed by this(#304) PR.

**Comment by Abhishek-kumar09** [comment]
What are the different alignment issues you are fixing?

**Comment by Abhijay007** [comment]
> What are the different alignment issues you are fixing?

***post something UI:***

before  |  after    |    
:-------------------------:|:-------------------------:|
post alignment|post alignment after|


Referral page UI :

before   |  after    |    
:-------------------------:|:-------------------------:|
before referral|after|

Peer page:

befor   |  after    |    
:-------------------------:|:-------------------------:|
before peer|after peer|

***on web:***

***before***
alignemt post something

***after***

fixed align

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Awesome, lots of fixes.

The changes in referral and coupons has introduced horizontal instability on smaller screens.Please revert

**Comment by Abhijay007** [comment]
Ok

File: src/pages/ReferralAndCouponCodes/index.js [hunk 2]
```diff
@@ -16,6 +16,13 @@ const useStyles = makeStyles((theme) => ({
   expanded: {
     flexGrow: 1,
   },
+  mainTitle: {
+    display: "flex",
+    marginTop: theme.spacing(2.1),
+    marginRight: theme.spacing(3),
+    alignItems: "center",
+    justifyContent: "space-between",
+  },
```
Comment (adarsh-technocrat) [inline src/pages/ReferralAndCouponCodes/index.js:25]: Hey, @Abhijay007 I don't think making the cancel button align with the title is a good idea you can revert this change :)

File: src/pages/ReferralAndCouponCodes/index.js [hunk 3]
```diff
@@ -53,9 +60,9 @@ function ReferralAndCoupons() {
   const history = useHistory();
   return (
     <>
-      <AppBar position="static" className={classes.appBar}>
-        <Toolbar variant="dense">
-          <Box className={classes.expanded} />
+      <Box className={classes.root}>
+        <div className={classes.mainTitle}>
+          <Typography variant="h2">Refer a friend</Typography>
```
Comment (adarsh-technocrat) [inline src/pages/ReferralAndCouponCodes/index.js:65]: same here @Abhijay007  you can revert this change :)

File: src/pages/ReferralAndCouponCodes/index.js [hunk 4]
```diff
@@ -64,10 +71,7 @@ function ReferralAndCoupons() {
           >
             <ClearIcon />
           </IconButton>
-        </Toolbar>
-      </AppBar>
-      <Box className={classes.root}>
-        <Typography variant="h2">Refer a friend</Typography>
+        </div>
```
Comment (adarsh-technocrat) [inline src/pages/ReferralAndCouponCodes/index.js:74]: here as well :)

File: src/pages/ReferralAndCouponCodes/index.js (hunk 2 above)
Comment (Abhijay007) [reply src/pages/ReferralAndCouponCodes/index.js:25]: @adarsh-technocrat can you please elaborate more on why it is not a good idea to align the cancel button with the title, 
I think there is nothing wrong in aligning the cancel button with the title as it was already in the same `div` I just did some minor changes to that div by adding a separate class. And in all other pages of the site as well the title of the page is in `Appbar`  along with goBack and navigate button :)

Comment (adarsh-technocrat) [reply src/pages/ReferralAndCouponCodes/index.js:25]: > @adarsh-technocrat can you please elaborate more on why it is not a good idea to align the cancel button with the title

oooh Okay! @Abhijay007 I think you are thinking that the title need's to be part of the `app Bar`  that's why you made that cancel button align to the title right! but actually, that was the purpose of the design that was not an alignment issue! 

Adding the Cancel button to the  `Material AppBar` itself makes it responsive so I don't think this will make any issue!
```javascript
 <AppBar position="static" className={classes.appBar}>
        <Toolbar variant="dense">
          <Box className={classes.expanded} />
          <IconButton
            edge="end"
            className={classes.clearIcon}
            color="inherit"
            onClick={() => history.goBack()}
          >
            <ClearIcon />
          </IconButton>
        </Toolbar>
      </AppBar>
```

Comment (adarsh-technocrat) [reply src/pages/ReferralAndCouponCodes/index.js:25]: > And in all other pages of the site as well the title of the page is in `Appbar` along with goBack and navigate button :)

 @Abhijay007  Actually it's not a mandatory thing to sync every route page with the same format.  We can change the format for some routes like this section just the thing is it should look good 😊.

Comment (Abhijay007) [reply src/pages/ReferralAndCouponCodes/index.js:25]: > oooh Okay! @Abhijay007 I think you are thinking that the title need's to be part of the `app Bar` that's why you made that cancel button align to the title right! but actually, that was the purpose of the design that was not an alignment issue!
> 
> Adding the Cancel button to the `Material AppBar` itself makes it responsive so I don't think this will make any issue!
> 

yes

Comment (Abhijay007) [reply src/pages/ReferralAndCouponCodes/index.js:25]: > > And in all other pages of the site as well the title of the page is in `Appbar` along with goBack and navigate button :)
> 
> @Abhijay007 Actually it's not a mandatory thing to sync every route page with the same format. We can change the format for some routes like this section just the thing is it should look good 😊.

Ya ! as the horizontal instability issue for smaller devices is resolved I think it is fine to implement this 😊 .

Comment (adarsh-technocrat) [reply src/pages/ReferralAndCouponCodes/index.js:25]: So @Abhijay007 you can revert the changes 😊

Comment (Abhijay007) [reply src/pages/ReferralAndCouponCodes/index.js:25]: > So @Abhijay007 you can revert the changes 😊

Can I revert the changes after knowing @Abhishek-kumar09 thoughts?

Comment (adarsh-technocrat) [reply src/pages/ReferralAndCouponCodes/index.js:25]: > Can I revert the changes after knowing @Abhishek-kumar09 thoughts?

Sure no worries  :)

Comment (Abhishek-kumar09) [reply src/pages/ReferralAndCouponCodes/index.js:25]: Referral page is more like a paper where there is coupon code and glossy design, It is not like the usual page and more like a special page. 

I was comapring https://deploy-preview-296--edu-client.netlify.app/referralAndCoupons and
https://edu-client.netlify.app/referralAndCoupons

You have lowered the cross button and made the cross and title on same level, which is fine but initial design is looking better in terms of the purpose and theme though there are not much difference, and I may be talking garbage. 
Also the design was approved by few mentors so we can leave it as it is for now. Though it's not a big deal and thanks for your efforts.

Comment (Abhijay007) [reply src/pages/ReferralAndCouponCodes/index.js:25]: > Referral page is more like a paper where there is coupon code and glossy design, It is not like the usual page and more like a special page.
> 
> I was comapring https://deploy-preview-296--edu-client.netlify.app/referralAndCoupons and
> https://edu-client.netlify.app/referralAndCoupons
> 
> You have lowered the cross button and made the cross and title on same level, which is fine but initial design is looking better in terms of the purpose and theme though there are not much difference, and I may be talking garbage.
> Also the design was approved by few mentors so we can leave it as it is for now. Though it's not a big deal and thanks for your efforts.

ok, no problem I will revert the changes 😊 , thanks @adarsh-technocrat @Abhishek-kumar09 for suggestions.

**Review by Abhishek-kumar09 — APPROVED** [review]
